### PR TITLE
[EPMRPP-41423] Data store refactoring

### PR DESCRIPTION
**Pull request by ihar-kahadouski**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/binary/CreateLogAttachmentService.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * @author <a href="mailto:ihar_kahadouski@epam.com">Ihar Kahadouski</a>
+ */
+@Service
+@Transactional
+public class CreateLogAttachmentService {
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/binary/CreateLogAttachmentService.java]: do we really need separate service for that?

Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/binary/CreateLogAttachmentService.java]: Yes, it should execute in one separate transaction

Comment (avarabyeu) [reply src/main/java/com/epam/ta/reportportal/binary/CreateLogAttachmentService.java]: would placing this stuff into separate method (within existing handler) and marking it with `@Transactional(requiresNew)` solve the issue?

Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/binary/CreateLogAttachmentService.java]: No. Self-invocation will not start transaction even if it market with annotation

File: src/main/java/com/epam/ta/reportportal/binary/DataStoreService.java [hunk 2]
```diff
@@ -45,6 +61,12 @@
 
 	private static final Logger LOGGER = LoggerFactory.getLogger(DataStoreService.class);
 
+	private static final String ROOT_USER_PHOTO_DIR = "users";
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/binary/DataStoreService.java]: Let's try to decompose all logic following single responsibility principle.
we would have UserBinaryDataHandler, AttachmentHandler that uses a common DataStoreService

**Pull request merged**